src/reactapp/src/components/paypalExpress/hooks/usePaypalExpress.js

```diff
@@ -0,0 +1,138 @@
+import { useCallback, useContext, useEffect, useState } from 'react';
+import AppContext from '../../../context/App/AppContext';
+import createCustomerToken from '../../../api/paypalExpress/createCustomerToken';
+import { _objToArray } from '../../../utils';
+import { __ } from '../../../i18n';
+import setPaymentMethodPaypalExpress from '../../../api/paypalExpress/setPaymentMethod';
+import placeOrder from '../../../api/cart/placeOrder';
+import useShippingMethodCartContext from '../../shippingMethod/hooks/useShippingMethodCartContext';
+import LocalStorage from '../../../utils/localStorage';
+import { config } from '../../../config';
+import usePaymentMethodCartContext from '../../paymentMethod/hooks/usePaymentMethodCartContext';
+import useBillingAddressCartContext from '../../billingAddress/hooks/useBillingAddressCartContext';
+
+export default function usePaypalExpress() {
+  const [, appActions] = useContext(AppContext);
+  const { setErrorMessage, setPageLoader } = appActions;
+  const { selectedMethod } = useShippingMethodCartContext();
```

Comment:
`useShippingMethodCartContext`, `useBillingAddressCartContext` & `usePaymentMethodCartContext` - these should not be used here. we created those custom hooks to use inside `shippingMethod`, `billingAddress`, `paymentMethod` respectively. So the thumb rule here is, we need to create custom hooks for each section so that it serves them locally. This way, the code is separate and we can easily remove the code chunks (if in case). It also helps to avoid the growing of the custom hook. 

So here we need to define custom hooks in `src/reactapp/src/components/paypalExpress/hooks/` directory. The name should be `usePaypalExpressCartContext` and then you need to prepare data in that custom hook from the `CartContext`. 

So basically `useShippingMethodCartContext`, `useBillingAddressCartContext` & `usePaymentMethodCartContext` custom hooks are collecting data from the `CartContext`. So our `usePaypalExpressCartContext` should essentially replace three of them

Reply:
Done